packages/connectivity/src/scp-cf/destination/destination-cache.spec.ts

```diff
@@ -317,6 +293,105 @@ describe('destination cache', () => {
+    it('disables the cache per default', async () => {
+      destinationCache.cacheRetrievedDestination(
+        { user_id: 'user', zid: 'tenant' },
+        destinationOne,
+        IsolationStrategy.Tenant
+      );
+      await expect(
+        getDestination({ destinationName: destName })
+      ).rejects.toThrowError(/Failed to fetch \w+ destinations./);
+    });
+
+    it('uses cache with isolation strategy Tenant if not JWT is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(providerServiceToken),
+        destinationOne,
+        IsolationStrategy.Tenant
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        useCache: true
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('uses cache with isolation strategy UserTenant if JWT is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(subscriberUserJwt),
+        destinationOne,
+        IsolationStrategy.Tenant_User
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        useCache: true,
+        jwt: subscriberUserJwt,
+        iasToXsuaaTokenExchange: false
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('enables cache if isolation strategy Tenant is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(providerServiceToken),
+        destinationOne,
+        IsolationStrategy.Tenant
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        isolationStrategy: IsolationStrategy.Tenant,
+        iasToXsuaaTokenExchange: false
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+    it('enables cache if isolation strategy TenantUser is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(subscriberUserJwt),
+        destinationOne,
+        IsolationStrategy.Tenant_User
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        isolationStrategy: IsolationStrategy.Tenant_User,
+        jwt: subscriberUserJwt,
+        iasToXsuaaTokenExchange: false
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('ignores cache if isolation requires user JWT but the JWT is not provided', async () => {
+      const logger = createLogger('destination-cache');
+      const warn = jest.spyOn(logger, 'warn');
+
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(subscriberUserJwt),
+        destinationOne,
+        IsolationStrategy.Tenant_User
+      );
+      await expect(
+        getDestination({
+          destinationName: destName,
+          isolationStrategy: IsolationStrategy.Tenant_User,
+          jwt: subscriberServiceToken,
+          iasToXsuaaTokenExchange: false
+        })
+      ).rejects.toThrowError(/Failed to fetch instance destinations./);
```

Comment:
I believe `getDestination` in this case will ultimately call `fetchInstanceDestinations` and `fetchSubaccountDestinations` in parallel and depending on which call fails first might lead to a different error. (`Failed to fetch subaccount destinations.`)